### PR TITLE
chore: move to go 1.18 and other library updates and minor fixes

### PR DESCRIPTION
**Pull request by ajayk**
- upgarde secrets-store-csi-driver to v1.1.1
- update to k8s version v0.23.5
- move flag.Parse() before the logger
- changes to the klog Json Logger

Signed-off-by: ajayk <ajaykemparaj@gmail.com>

### CODE REVIEW

File: tools/go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/GoogleCloudPlatform/secrets-store-csi-driver-provider-gcp/tools
 
-go 1.17
+go 1.18
```
Comment (tam7t) [inline tools/go.mod:3]: It looks like `honnef.co/go/tools` needs to be updated to v0.3.0 to support go1.18

**Review by tam7t — CHANGES_REQUESTED** [review]
Thanks for your contribution!

File: tools/go.mod (hunk 1 above)
Comment (ajayk) [reply tools/go.mod:3]: Thanks @tam7t  , i have updated few more libraries

Comment (ajayk) [reply tools/go.mod:3]: Fixed the static check on the test as well

Comment (ajayk) [reply tools/go.mod:3]: @tam7t i can rebase the commit if you need me to

**Review by ajayk — COMMENTED** [review]
suggested changes has been  completed

**Review by tam7t — APPROVED** [review]

**Pull request merged**